Cesium3DTiles/include/Cesium3DTiles/CartographicSelection.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include "Cesium3DTiles/Library.h"
+#include "CesiumGeospatial/GlobeRectangle.h"
+#include <glm/vec2.hpp>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace Cesium3DTiles {
+
+/**
+ * @brief A 2D polygon that can be rasterized onto {@link Tileset} objects.
+ */
+class CESIUM3DTILES_API CartographicSelection final {
+public:
+  /**
+   * @brief Constructs a 2D polygon that can be rasterized onto {@link Tileset}
+   * objects.
+   *
+   * @param targetTextureName The desired name of the target texture to be
+   * rasterized. The tile glTFs containing this polygon will carry a texture id
+   * in their model.extras by the name of CUSTOM_MASK_<targetTextureName>.
+   * @param polygon An array of longitude-latitude points in WGS84 defining the
+   * perimeter of the 2D polygon.
+   * @param isForCulling Whether this selection is going to be used strictly
+   * for culling. If it will only be used for culling inside the polygon, the
+   * streaming can be optimized by avoiding loading tiles that are entirely
+   * within the culled area.
+   */
+  CartographicSelection(
+      const std::string& targetTextureName,
+      const std::vector<glm::dvec2>& polygon,
+      bool isForCulling);
+
+  /**
+   * @brief Returns the target texture name that the polygon will be rasterized
+   * into. The tile glTFs containing this polygon will carry a texture id in
+   * their model.extras by the name of CUSTOM_MASK_<targetTextureName>.
+   *
+   * @return The target texture name.
+   */
+  constexpr const std::string& getTargetTextureName() const {
+    return this->_targetTextureName;
+  }
+
+  /**
+   * @brief Returns the longitude-latitude vertices in WGS84 that define the
+   * perimeter of the selected polygon.
+   *
+   * @return The perimeter vertices in WGS84 longitude-latitude.
+   */
+  constexpr const std::vector<glm::dvec2>& getVertices() const {
```

Comment:
I maybe wrong on this one, but I think `constexpr` doesn't have effect for this class because up to C++17, std::string and std::vector can't be constructed at compile time. C++20 will allow them, but I haven't had any luck to find the compilers that can init the string at compile time (something like `constexpr std::string s = "Hello World"`, this can only be done with `std::string_view` at the moment). So `CartographicSelection` can't be constructed at compile time